src/mono/wasm/runtime/method-calls.ts

```diff
@@ -541,18 +541,18 @@ export function mono_wasm_invoke_js(code: MonoString, is_exception: Int32Ptr): M
     if (code === MonoStringNull)
         return MonoStringNull;
 
-    const js_code = conv_string(code);
+    const js_code = conv_string(code)!;
 
     try {
-        const closure = {
-            Module, MONO, BINDING, INTERNAL
+        const closedEval = function (Module: EmscriptenModule, MONO: any, BINDING: any, INTERNAL: any, code: string) {
+            return eval(code);
         };
-        const fn_body_template = `const {Module, MONO, BINDING, INTERNAL} = __closure; const __fn = function(){ ${js_code} }; return __fn.call(__closure);`;
-        const fn_defn = new Function("__closure", fn_body_template);
-        const res = fn_defn(closure);
+        let res = closedEval(Module, MONO, BINDING, INTERNAL, js_code);
         Module.setValue(is_exception, 0, "i32");
         if (typeof res === "undefined" || res === null)
             return MonoStringNull;
+
+        res = res.toString();
```

Comment:
if res is null or undefined, this will fail. i'll check what the appropriate solution for this is

Reply:
Isn't it covered two lines above? 
I have also added a test, where the result of the JS expression is `null`.

Reply:
ok, this is actually fine, I think. We already check for the type up above, the only thing this should fail for is Symbol, which already was broken

Reply:
Yeah, I misread the check above initially. Symbol is the only problem type here, and you don't need to worry about it (Symbol is generally broken even without our help)